bson/codec_options.py

```diff
@@ -32,10 +33,41 @@ def _raw_document_class(document_class):
     return marker == _RAW_BSON_DOCUMENT_MARKER
 
 
+class TypeRegistry(object):
+    def __init__(self, encoder_map=None, decoder_map=None):
+        self.__encoder_map = encoder_map or {}
+        self.__decoder_map = decoder_map or {}
```

Comment:
I think the registry should make a copy of each map here. That way if the app makes changes to the maps after creating the TypeRegistry, the changes will not be reflected in the TypeRegistry.